Lib/importlib/resources.py

```diff
@@ -0,0 +1,303 @@
+import os
+import tempfile
+
+from . import abc as resources_abc
+from builtins import open as builtins_open
+from contextlib import contextmanager, suppress
+from importlib import import_module
+from importlib.abc import ResourceLoader
+from io import BytesIO, TextIOWrapper
+from pathlib import Path
+from types import ModuleType
+from typing import Iterator, Optional, Set, Union   # noqa: F401
```

Comment:
Should we drop the directive comment?

Or better yet, do we even need the `typing`-related imports with the "lazy" type hint changes in Python 3.7?

Reply:
Has that landed yet?  I tried removing them (except for `Union`, which we'd still need because of the `Package` and `Resource` definitions).  Python was not happy.  So I guess we can address that again later.

I could remove the `#noqa` directive, but it makes my editor happier, so I think I'll leave them in for now.